pkgs/development/compilers/swi-prolog/default.nix

```
@@ -25,6 +28,7 @@ stdenv.mkDerivation {
   buildInputs = [ cacert git cmake gmp readline openssl
     libarchive libyaml db pcre libedit libossp_uuid
     zlib pkgconfig ]
+  ++ (if gui then guiInputs else [])
```

generally i see a "withXXX" prefix used when passing flags to an expression, that way it's not confused with a package.
```suggestion
  ++ lib.optionals withGui guiInputs
```